### PR TITLE
feat: add dropdown for topics in Explore

### LINKED ISSUE #533: Constrain available topics to predefined list
> From talking with @bdougie there would be an "official" list of around 7 topics, but I think the goal is to always be specific and not be a wildcard search.

On top of my head, the topics would be ideal defaults
- JavaScript
- Rust
- Python
- ml (machine-learning)
- ai
- React

_Originally posted by @bdougie in https://github.com/open-sauced/insights/issues/528#issuecomment-1276506004_

### PR DESCRIPTION
**Pull request by Deadreyo**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
- Adds a new (atom) component for selector.
- Adds a new (molecule) component for filter card with selection.
- Changes the `camelCaseNaming` lib into a more general name-formatting function
- Implements the new filter card with selection component to enable selecting topics from a pre-defined list in Explore.

## Follow-up tasks
A new issue should be made to re-use the new `selector` component in the existing `superlative-selector`

## Related Tickets & Documents

closes #533 
## Mobile & Desktop Screenshots/Recordings
image
image

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by bdougie — CHANGES_REQUESTED** [review]
This section is meant to share totals. If that is not possible, could we delete

Screen Shot 2023-02-15 at 2 32 29 PM

File: lib/hooks/useTopicOptions.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const useTopicOptions = () => {
```
Comment (brandonroberts) [inline lib/hooks/useTopicOptions.ts:1]: By convention, this should not be prefixed with `use` as it is not a hook, but a function that returns data. It should also be moved to `lib/utils`

```suggestion
const getTopicOptions = () => {
```

File: components/molecules/FilterCardSelect/filter-card-select.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+const FilterCardSelect: React.FC<FilterCardSelectProps> = ({ selected: filterName, icon = "topic", options, handleFilterClick }) => {
+
+  const ref = useRef<HTMLDivElement>(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const toggleFilter = () => {
+    setIsOpen(!isOpen);
+  };
+
+  useEffect(() => {
+    const checkIfClickedOutside = (e: any) => {
```
Comment (brandonroberts) [inline components/molecules/FilterCardSelect/filter-card-select.tsx:45]: This should not be `e: any`. 

```suggestion
    const checkIfClickedOutside = (e: MouseEvent) => {
```

**Comment by brandonroberts** [comment]
> This section is meant to share totals PRs represented in the topic. If that is not possible, could we delete circle UI element.
> 
> Screen Shot 2023-02-15 at 2 32 29 PM

I think we should remove it for now

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Comment by Deadreyo** [comment]
> This section is meant to share totals PRs represented in the topic. If that is not possible, could we delete circle UI element.
> 
> Screen Shot 2023-02-15 at 2 32 29 PM

deleted

**Review by bdougie — APPROVED** [review]

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**